src/Adapter/MSTest.TestAdapter/Execution/TestExecutionManager.cs

```diff
@@ -471,7 +471,11 @@ private async Task ExecuteTestsInSourceAsync(IEnumerable<TestCase> tests, IRunCo
 
                 try
                 {
-                    await Task.WhenAll(tasks);
+                    // Ensure we can abandon currently running tasks on cancellation, rather than waiting for them to complete.
+                    // They will still run on background but we won't await them.
+                    var runCancelled = new TaskCompletionSource<object>();
+                    _testRunCancellationToken?.Register(() => runCancelled.TrySetCanceled());
+                    await Task.WhenAny(Task.WhenAll(tasks), runCancelled.Task);
```

Comment:
Cancellation token _testRunCancellationToken is not a CancellationToken, so we cannot use `Task.WhenAll(tasks, ct)` here.